cypress/integration/smoke/balances.spec.js

```diff
@@ -0,0 +1,196 @@
+  // Token balance regex
+  const tokenRegex = new RegExp(`([0-9]{1,3},)*[0-9]{1,3}.[0-9]{2,5}`)
+
+  before(() => {
+    // Open the Safe used for testing
+    cy.visit(`${TEST_SAFE}/balances`)
+    cy.contains('a', 'Accept selection').click()
+  })
+
+  describe('should have different tokens', () => {
+    it(`should have ${ASSETS_LENGTH} entries in the table`, () => {
+      cy.get(balanceRowTestId).should('have.length', ASSETS_LENGTH)
+    })
+
+    it('should have Dai', () => {
+      const DAI_ROW_NUMBER = 0
+      // First row should be Dai
+      cy.get(balanceRowTestId).eq(DAI_ROW_NUMBER).contains('Dai')
+
+      // First row should have a image with alt text "Dai"
+      cy.get(balanceRowTestId).eq(DAI_ROW_NUMBER).get('img[alt="Dai"]').should('exist')
+
+      // Link to show details
+      cy.get(balanceRowTestId).eq(DAI_ROW_NUMBER).get('a[aria-label="Show details on Etherscan"]').should('exist')
+
+      // Balance should end with DAI
+      cy.get(balanceRowTestId).eq(DAI_ROW_NUMBER).find('td').eq(TOKEN_AMOUNT_COLUMN).contains('DAI')
+    })
+
+    it('should have Wrapped Ether', () => {
+      const WETH_ROW_NUMBER = 1
+
+      cy.get(balanceRowTestId).contains('Wrapped Ether')
+
+      // First row should have a image with alt text "Wrapped Eth"
+      cy.get(balanceRowTestId).eq(WETH_ROW_NUMBER).get('img[alt="Wrapped Ether"]').should('exist')
+
+      // Link to show details
+      cy.get(balanceRowTestId).eq(WETH_ROW_NUMBER).get('a[aria-label="Show details on Etherscan"]').should('exist')
+
+      // Balance should end with WETH
+      cy.get(balanceRowTestId).eq(WETH_ROW_NUMBER).find('td').eq(TOKEN_AMOUNT_COLUMN).contains('WETH')
+    })
+
+    it('should have USD Coin', () => {
+      const USD_ROW_NUMBER = 3
+
+      cy.get(balanceRowTestId).contains('USD Coin')
+
+      // First row should have a image with alt text "USD Coin"
+      cy.get(balanceRowTestId).eq(USD_ROW_NUMBER).get('img[alt="USD Coin"]').should('exist')
+
+      // Link to show details
+      cy.get(balanceRowTestId).eq(USD_ROW_NUMBER).get('a[aria-label="Show details on Etherscan"]').should('exist')
+
+      // Balance should end with USDC
+      cy.get(balanceRowTestId).eq(USD_ROW_NUMBER).find('td').eq(TOKEN_AMOUNT_COLUMN).contains('USDC')
+    })
+  })
+
+  describe('values should be formatted as per locale', () => {
+    it('should have Token balance formated as per tokenRegex', () => {
+      // Balance should comply with the tokenRegex
+      cy.get(assetsTableContainer)
+        .find(balanceRowTestId)
+        .eq(1)
+        .find('td')
+        .eq(TOKEN_AMOUNT_COLUMN)
+        .should(($div) => {
+          const text = $div.text()
+          expect(text).to.match(tokenRegex)
+        })
+    })
+
+    it('should have Fiat balance formated as per fiatRegex', () => {
+      // Balance should comply with the fiatRegex
+      cy.get(assetsTableContainer)
+        .find(balanceRowTestId)
+        .eq(1)
```

Comment:
`WETH_ROW_NUMBER`

Reply:
You're only checking formatting of WETH, can't we test it for each row in a loop?